src/bin/nydus-image/core/blob.rs

```diff
@@ -36,16 +36,23 @@ impl Blob {
                 let (inodes, prefetch_entries) = blob_ctx
                     .blob_layout
                     .layout_blob_simple(&ctx.prefetch, nodes)?;
+                let mut has_chunk = false;
                 for (idx, inode) in inodes.iter().enumerate() {
                     let node = &mut nodes[*inode];
                     let size = node
                         .dump_blob(ctx, blob_ctx, blob_index, chunk_dict)
                         .context("failed to dump blob chunks")?;
+                    if size > 0 {
+                        has_chunk = true
+                    }
                     if idx < prefetch_entries {
                         blob_ctx.blob_readahead_size += size;
                     }
                 }
-                self.dump_meta_data(blob_ctx)?;
+                // Dump is only required if there is chunk in the blob
+                if has_chunk {
+                    self.dump_meta_data(blob_ctx)?;
```

Comment:
Better to check `blob_ctx.compressed_blob_size` or `blob_ctx.chunk_count` in `dump_meta_data`, so that we can reuse this logic for stargz build in future.

Reply:
fixed